test/jdk/java/net/InetAddress/HostsFileOrderingTest.java

```diff
@@ -84,16 +88,18 @@ public void generateHostsFile() throws Exception {
      */
     @Test
     public void testOrdering() throws Exception {
-        String [] resolvedAddresses = Arrays.stream(InetAddress.getAllByName("hostname.test.com"))
-                .map(InetAddress::getHostAddress).toArray(String[]::new);
-        String [] expectedAddresses = getExpectedAddressesArray();
-
-        if (Arrays.deepEquals(resolvedAddresses, expectedAddresses)) {
-            System.err.println("Test passed: The expected list of IP addresses is returned");
-        } else {
-            System.err.printf("Expected addresses:%n%s%n", Arrays.deepToString(expectedAddresses));
-            System.err.printf("Resolved addresses:%n%s%n", Arrays.deepToString(resolvedAddresses));
-            Assert.fail("Wrong host resolution result is returned");
+        if (IPSupport.hasIPv6()) {
```

Comment:
Hello Matthias, unlike changes to some other tests, this skips the entire test method if there's no IPv6 support. So perhaps add a System.out message that this test is skipped? I checked if `testng` has `assumeTrue` kind of API like junit, but I couldn't find any, so just returning by printing a System.out message might be enough.

Reply:
Some of the test runs should work whether the machine has IPv6 or not. Maybe this test should be more selective with respect to when the test should be skipped. For instance at line 47-48: 
```
 * @run testng/othervm -Djdk.net.hosts.file=TestHostsFile.txt
 * -Djava.net.preferIPv4Stack=true -Djava.net.preferIPv6Addresses=false HostsFileOrderingTest
```
well - that should work - and only return IPv4 addresses.

On second thought - I suspect it's `getExpectedAddressesArray()` that should be fixed to take into account IPSupport.hasIPv6() instead of doing it here.